### PR TITLE
[CBRD-22950] Slave ack on consume 

### PR DESCRIPTION
**Pull request by radudoros**
http://jira.cubrid.org/browse/CBRD-22950

Comprising changes:
1. Send ack on log_consumer::fetch_stream_entry.
2. Fix create daemon with contexts (for notify_ack) & use destroy daemon calls corresponding to their create one's (no_ctxt create -> no_ctxt destroy 
3. Remove most of the replication-related things from SA's CmakeLists.txt.

### CODE REVIEW

File: src/replication/replication_slave_node.cpp [hunk 1]
```diff
@@ -112,33 +108,26 @@ namespace cubreplication
...
-    g_instance->m_transfer_receiver = new cubstream::transfer_receiver (std::move (srv_chn), *g_instance->m_stream,
-	start_position);
 
-    g_instance->m_control_channel_sender = new cubreplication::slave_control_channel (std::move (control_chn));
-#endif
+    g_instance->m_lc->set_ctrl_chn (new cubreplication::slave_control_channel (std::move (control_chn)));
 
-    return error;
+    g_instance->m_transfer_receiver = new cubstream::transfer_receiver (std::move (srv_chn), *g_instance->m_stream,
+	start_position);
```
Comment (ribram) [inline src/replication/replication_slave_node.cpp]: add `return NO_ERROR;`

File: src/replication/log_consumer.hpp [hunk 2]
```diff
@@ -85,6 +87,8 @@ namespace cubreplication
     private:
       std::queue<stream_entry *> m_stream_entries;
 
+      std::unique_ptr<slave_control_channel> m_ctrl_chn;
```
Comment (razvanradulescu) [inline src/replication/log_consumer.hpp]: For me, it looks more natural that the slave_node instance is the owner of the slave_control_channel object, and the log_consumer uses a reference to it.
Generally, we may configure between several contexts to send the stream ACK : when stream position reaches the slave or when it is flushed to disk. With current design, we are required to change "owner" of the slave_control_channel either to log_consumer or stream_file (or the object responsible with flushing).
Also, from logical point of view, log_consumer handles only pure replication data, the control data seems another domain. There is the idea to change the name of log_consumer as replication_stream_consumer.

File: src/replication/log_consumer.cpp [hunk 3]
```diff
@@ -317,13 +319,13 @@ namespace cubreplication
       }
 
     entry = se;
+    m_ctrl_chn->send_ack (entry->get_stream_entry_end_position ());
```
Comment (razvanradulescu) [inline src/replication/log_consumer.cpp]: should be stream_entry :: start_position (or at least data_start_position)
At this point, in stream we have only the stream entry header, the contents have not yet reached the slave.

File: src/replication/log_consumer.hpp (hunk 2 above)
Comment (eseokoh) [reply src/replication/log_consumer.hpp]: I'd like to merge the patch and continue discussion. Please find http://jira.cubrid.org/browse/CBRD-22996

**Pull request merged**